### DISCUSSION TITLE
Ruby kodlarını başka dillere entegre etmenin en iyi yolu nedir ?

### DISCUSSION THREAD

**Discussion by BunyaminAkdeniz** [19/classroom#47] [Anonim] [answered]
Elimde bir dart dosyası var. Çalıştırdığımda bir hata oluşursa ekrana basıyor.
```dart
void main(List<String> args) {
  int sayi = 3;
  try {
    sayi / 3;
  } catch (e) {
    print("hata meydana geldi " + e.toString());
  } finally {
    ..
  }
  print("devam eden kod");
  ..
}
```
Bu aşamada hata çıktılarını yazmadan önce ruby'de işlemden geçirdikten sonra kaydetmek istiyorum. Program her çalıştığında çıktı ruby kodumda işlem görüp kaydedilecek. Terminal dışında otomatik bir şekilde bu program çalıştığında çıktısının ruby koduma yönlendirmesini nasıl sağlayabilirim ?

**Comment by roktas** [comment]
Bunun pek çok yolu var.  Senaryo tam olarak nedir?  Bu bir web uygulaması mı, terminalde çalışacak şekilde kurgulanan bir uygulama mı?  Ayrıntı vermen lazım.

**Reply by BunyaminAkdeniz** [reply]
Android ve iOS cihazlarda çalışacak bir flutter uygulaması. Senaryom çıkan hata mesajlarının sadece hata kodunu loglayıp, databaseme kaydetmek. Loglama aşamasında ki filtreyi ve databaseye push işlemini Ruby ile yazdım. (Bu işlemler dart dili ile yapılabilir. Sadece öğrenmek için yapıyorum)

**Comment by roktas** [comment, marked as answer]
İki farklı dilde yazılmış programı konuşturmanın en basit (ama senaryoya göre maliyetli ve/veya çirkin olabilecek) yolu IPC, yani bir programın diğer programı bir proses yoluyla çalıştırması (exec vs).  Gördüğüm kadarıyla bunu Dart'ta şöyle yapıyorsun: https://api.flutter.dev/flutter/dart-io/Process-class.html (Ruby programını uygun bir çalıştırılabilir dosyaya dönüştüreceksin varsayımıyla)

Ama bana kalırsa bu da çirkin bir yol olur.  Flutter'ın log mekanizmalarını incelemelisin.  Örneğin ilk bakışta custom log filtreleri kullanılabilir gibi görünüyor.  Eğer böyleyse tüm loglamaları Flutter log alt yapısıyla yap ve yukarıda açıkladığım kodu filtre metoduna koy.  Ama eminim ki bu çözümden de daha şık olan bir yol bulunabilir.

**Reply by BunyaminAkdeniz** [reply]
Teşekkür ederim hocam. Araştıracağım
